src/frontend/src/lib/components/guard/AddressGuard.svelte

```diff
@@ -1,29 +1,49 @@
 <script lang="ts">
 	import { addressStore } from '$lib/stores/address.store';
-	import { isNullish } from '@dfinity/utils';
-	import { certifyEthAddress } from '$lib/services/address.services';
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { certifyBtcAddressMainnet, certifyEthAddress } from '$lib/services/address.services';
 	import { warnSignOut } from '$lib/services/auth.services';
-	import { ETHEREUM_TOKEN_ID } from '$env/tokens.env';
+	import {
+		btcAddressMainnet,
+		btcAddressMainnetData,
+		btcAddressMainnetNotCertified,
+		ethAddress,
+		ethAddressData,
+		ethAddressNotCertified
+	} from '$lib/derived/address.derived';
+	import { NETWORK_BITCOIN_ENABLED } from '$env/networks.btc.env.js';
+	import type { ResultSuccess } from '$lib/types/utils';
 
 	const validateAddress = async () => {
-		const ethAddressData = $addressStore?.[ETHEREUM_TOKEN_ID];
-
-		if (isNullish(ethAddressData)) {
+		if (isNullish($btcAddressMainnetData) && isNullish($ethAddressData)) {
 			// No address is loaded, we don't have to verify it
 			return;
 		}
 
-		const { certified, data: ethAddress } = ethAddressData;
-
-		if (certified === true) {
-			// The address is certified, all good
+		if ((!NETWORK_BITCOIN_ENABLED || !$btcAddressMainnetNotCertified) && !$ethAddressNotCertified) {
+			// The addresses are certified, all good
 			return;
 		}
 
-		const { success, err } = await certifyEthAddress(ethAddress);
+		const results = await Promise.all([
```

Comment:
Reading the code again. I'm a bit concerned about triggering too many update calls. Isn't there a race condition here in which we might trigger twice for example `certifyEthAddress`? If yes, would it be maybe safer to split both checks?

```
$: $ethAddressSomething, (async () => await validateEthAddress())();
$: $btcAddressSomething, (async () => await validateBtcAddress())();
```

Of course it there is no such issue we do not need it. I'm just being really precaucious here.

Reply:
ah! understood you point, make sense! however, I put all addresses in the same store: `ethAddressSomething` and `btcAddressSomething` will always be derived from there and triggered together for any change in the addresses store...

Shall I change then and create different stores for different addresses? I can put this PR on hold and do that first in another PR

Reply:
@peterpeterparker WDYT of my comment above?

Reply:
> will always be derived from there and triggered together for any change in the addresses store...

I actually have a doubt. If you have a store and two derived states, derivedA and derivedB, are they both triggered when the store changes, or only when their values change?

Reply:
Ah but we do not have a condition here that would trigger only derivedA or derivedB.

Reply:
AH in any case, looking at the code again, there is something else that does not work neither.

`validateAddress` is triggered when `$addressStore` changes but within the function derived store are used. Therefore there is a chance that the derived store are not populated yet with the value derived. It's an almost impossible race condition but, it's one.

Reply:
So to answer your question, I feel like you are right, we might need different stores. I don't see right now another solution but, if you have an idea, I would be really happy to hear it!!!!

Regarding the race condition, we can then do something as following:

```
const validateAddress = async (data: OptionAddressData) => {
     // Same logic of the function as on main but using data parameter
}

$: $btcAddressMainnetData, (async () => await validateAddress($btcAddressMainnetData))();
$: $ethAddressMainnetData, (async () => await validateAddress($ethAddressMainnetData))();
```

Reply:
Yes, I don't see any other solution honestly. I will split that in several PR:

- PR #2207